cloud/pkg/controllermanager/edgeapplication/overridemanager/commandoverrider.go

```diff
@@ -33,9 +34,44 @@ func (o *CommandOverrider) ApplyOverrides(rawObj *unstructured.Unstructured, ove
 		}
 	}
 
+	// Apply node affinity based on TargetNodeLabelSelector
+	if len(overriders.TargetNodeLabelSelector.MatchLabels) > 0 {
+		if err := ApplyNodeAffinity(rawObj, overriders.TargetNodeLabelSelector); err != nil {
+			return fmt.Errorf("failed to apply node affinity: %v", err)
+		}
+	}
+
 	return nil
 }
 
+// ApplyNodeAffinity applies node affinity to the object based on the label selector
+func ApplyNodeAffinity(obj *unstructured.Unstructured, selector v1.LabelSelector) error {
+	affinity := map[string]interface{}{
+		"nodeAffinity": map[string]interface{}{
+			"requiredDuringSchedulingIgnoredDuringExecution": map[string]interface{}{
+				"nodeSelectorTerms": []interface{}{
+					map[string]interface{}{
+						"matchExpressions": []interface{}{},
+					},
+				},
+			},
+		},
+	}
+
+	for key, value := range selector.MatchLabels {
+		affinity["nodeAffinity"].(map[string]interface{})["requiredDuringSchedulingIgnoredDuringExecution"].(map[string]interface{})["nodeSelectorTerms"].([]interface{})[0].(map[string]interface{})["matchExpressions"] = append(
+			affinity["nodeAffinity"].(map[string]interface{})["requiredDuringSchedulingIgnoredDuringExecution"].(map[string]interface{})["nodeSelectorTerms"].([]interface{})[0].(map[string]interface{})["matchExpressions"].([]interface{}),
+			map[string]interface{}{
+				"key":      key,
+				"operator": "In",
+				"values":   []interface{}{value},
+			},
+		)
```

Comment:
Here, the processing performance is better with map, but the readability is worse. Is it better to use jsonpath?